src/rules/fileNameCasingRule.ts

```diff
@@ -92,15 +140,53 @@ export class Rule extends Lint.Rules.AbstractRule {
         }
     }
 
+    private static isValidRegExp(regExpString: string) {
+        try {
+            RegExp(regExpString);
+            return true;
+        } catch (e) {
+            return false;
+        }
+    }
+
+    private static findApplicableCasing(
+        fileBaseName: string,
+        fileNameCasings: FileNameCasings
+    ): Casing | null {
+        const applicableCasing = fileNameCasings.find(fileNameCasing => {
+            const fileNameMatch = fileNameCasing[0];
+            return (
+                Rule.isValidRegExp(fileNameMatch) && RegExp(fileNameMatch, "i").test(fileBaseName)
```

Comment:
...so if an invalid regular expression is provided, the rule will fail silently? 

We should prefer to fail verbosely for invalid configurations. Either remove the `isValidRegExp` check _(+1)_ or `console.warn` if it fails.

There are a lot of ways to provide invalid configurations, and it's near-impossible to guard against all of them. TSLint is moving towards standardized config options but it'll be a while before they're ready.

Reply:
Yeah, there already were cases like this in the current implementation, see [here](https://github.com/palantir/tslint/blob/master/src/rules/fileNameCasingRule.ts#L96) for example: if the mandatory casing wasn't provided simply return an empty array 🤷‍♂️
I would like to fail more verbosely as you suggested, hence my question that you already answered below.
Currently there don't seem to be any `console.warn`'s in the rules, so maybe there's a better way? In [objectLiteralSortKeys](https://github.com/palantir/tslint/blob/master/src/rules/objectLiteralSortKeysRule.ts#L103) for example an actual `Error` is thrown.
But I will go with what you guys suggest 🌝

Reply:
@JoshuaKGoldberg I just realized that it might be not a good idea to either throw an error or use console.warn, since this would happen for every linted file in the project. Using `showWarningOnce` on the other should work, some rules even use this already. Any other suggestions?

Reply:
Agreed. Filed #4280 to tackle this more broadly.

In the meantime, feel free to either use the `showWarningOnce` approach or not validate. IMO it shouldn't be each rule's responsibility to validate options _ideally_, since they already specify their format in their metadata... but since there isn't yet a method for that, it seems up to you.

Reply:
Thanks for the issue! (There's a small typo in my mention 😌)